cmd/joy/release_promote_filters_test.go

```diff
@@ -0,0 +1,222 @@
+	"github.com/nestoca/joy/internal/git/pr"
+	"github.com/nestoca/joy/internal/info"
+	"github.com/nestoca/joy/internal/links"
+	"github.com/nestoca/joy/internal/release/cross"
+	"github.com/nestoca/joy/internal/release/promote"
+	"github.com/nestoca/joy/internal/yml"
+	"github.com/nestoca/joy/pkg/catalog"
+)
+
+func TestReleasePromoteFilters(t *testing.T) {
+	type TestFile struct {
+		Path    string
+		Content string
+	}
+
+	type TestCrossRelease struct {
+		Name   string
+		Source *TestFile
+		Target *TestFile
+	}
+
+	cases := []struct {
+		Name         string
+		Args         []string
+		Releases     []TestCrossRelease
+		Expectations func(t *testing.T, files []*yml.File)
+	}{
+		{
+			Name: "all flag selects releases for update",
+			Args: []string{"--all"},
+			Releases: []TestCrossRelease{
+				{
+					Name:   "alpha",
+					Source: &TestFile{Path: "alpha-source.yaml", Content: "{spec: {version: 1.2.3}}"},
+					Target: &TestFile{Path: "alpha-target.yaml", Content: "{spec: {version: 1.0.0}}"},
+				},
+				{
+					Name:   "beta",
+					Source: &TestFile{Path: "beta-source.yaml", Content: "{spec: {version: 3.2.1}}"},
+					Target: &TestFile{Path: "beta-target.yaml", Content: "{spec: {version: 3.0.0}}"},
+				},
+			},
+			Expectations: func(t *testing.T, files []*yml.File) {
+				require.Len(t, files, 2)
+
+				require.Equal(t, "alpha-target.yaml", filepath.Base(files[0].Path))
+				require.Equal(t, "{spec: {version: 1.2.3}}\n", string(files[0].Yaml))
+
+				require.Equal(t, "beta-target.yaml", filepath.Base(files[1].Path))
+				require.Equal(t, "{spec: {version: 3.2.1}}\n", string(files[1].Yaml))
+			},
+		},
+		{
+			Name: "all flag with omit",
+			Args: []string{"--all", "--omit=alpha"},
+			Releases: []TestCrossRelease{
+				{
+					Name:   "alpha",
+					Source: &TestFile{Path: "alpha-source.yaml", Content: "{spec: {version: 1.2.3}}"},
+					Target: &TestFile{Path: "alpha-target.yaml", Content: "{spec: {version: 1.0.0}}"},
+				},
+				{
+					Name:   "beta",
+					Source: &TestFile{Path: "beta-source.yaml", Content: "{spec: {version: 3.2.1}}"},
+					Target: &TestFile{Path: "beta-target.yaml", Content: "{spec: {version: 3.0.0}}"},
+				},
+			},
+			Expectations: func(t *testing.T, files []*yml.File) {
+				require.Len(t, files, 1)
+				require.Equal(t, "beta-target.yaml", filepath.Base(files[0].Path))
+				require.Equal(t, "{spec: {version: 3.2.1}}\n", string(files[0].Yaml))
+			},
+		},
+		{
+			Name: "all flag with keep pre release",
+			Args: []string{"--all", "--keep-pre-release"},
+			Releases: []TestCrossRelease{
+				{
+					Name:   "pre-release-target",
+					Source: &TestFile{Path: "alpha-source.yaml", Content: "{spec: {version: 1.2.3}}"},
+					Target: &TestFile{Path: "alpha-target.yaml", Content: "{spec: {version: 1.0.0-some-feature}}"},
+				},
+				{
+					Name:   "nil-target",
+					Source: &TestFile{Path: "beta-source.yaml", Content: "{spec: {version: 3.2.1}}"},
+					Target: nil,
+				},
```

Comment:
Maybe just add a comment to explain why this nil target matters in this test? Otherwise, it may look like testing two different cases in the same test, whereas it's really related. Or maybe it could be broken down into two separate tests to make things clearer(?)